curriculum/challenges/english/25-front-end-development/quiz-javascript-fundamentals/66edcd875b0d91de1fbbb492.md

```diff
@@ -17,439 +17,532 @@ To pass the quiz, you must correctly answer at least 18 of the 20 questions belo
 
 ---
 
-Provide a set default parameters.
+String primitives are stored as objects internally.
 
 ---
 
-Destructure an object into parameters.
+JavaScript temporarily converts the string primitive to an array.
 
 #### --answer--
 
-Capture all arguments into one array.
+JavaScript temporarily wraps the string primitive in a String object.
 
 ### --question--
 
 #### --text--
 
-Which of the following is the correct syntax for the rest parameter in JavaScript?
+What will the following code log to the console?
+
+```js
+let stringArray = [8, 9, 10].toString();
+console.log(stringArray);
+```
 
 #### --distractors--
 
-`function myFunction(restParams)`
+`[8,9,10]`
 
 ---
 
-`function myFunction(restParams...)`
+`["8", "9", "10"]`
 
 ---
 
-`function myFunction([...restParams])`
+`"8", "9", "10"`
 
 #### --answer--
 
-`function myFunction(...restParams)`
+`"8,9,10"`
 
 ### --question--
 
 #### --text--
 
-What does destructuring allows you to do in JavaScript?
+How can you convert a string literal into a Number object?
 
 #### --distractors--
 
-Build complex objects.
+With the `Object()` constructor.
 
 ---
 
-Serialize objects.
+With the `.toNumber()` method.
 
 ---
 
-Merge arrays or objects.
+With the `.parseInt()` method.
 
 #### --answer--
 
-Extract properties from objects and elements from arrays.
+With the `Number()` constructor.
 
 ### --question--
 
 #### --text--
 
-Which of the following is a valid function name in JavaScript?
+What will `Number(true)` and `Number(false)` return, respectively?
 
 #### --distractors--
 
-`123function`
+`undefined` and `undefined`
 
 ---
 
-`function-name`
+`NaN` and `NaN`
 
 ---
 
-`function`
+`true` and `false`
 
 #### --answer--
 
-`functionName`
+`1` and `0`
 
 ### --question--
 
 #### --text--
 
-Which of these is NOT a good practice for naming variables and functions in JavaScript?
+What will `Number(undefined)` and `Number(null)` return, respectively?
 
 #### --distractors--
 
-Use camelCase notation.
+`NaN` and `NaN`
 
 ---
 
-Start variable names with a letter, $ or _.
+`0` and `0`
 
 ---
 
-Using meaningful names.
+`undefined` and `null`
 
 #### --answer--
 
-Using reserved keywords as variable names.
+`NaN` and `0`
 
 ### --question--
 
 #### --text--
 
-What is the primary purpose of a linter in JavaScript development?
+Which of the following is the best name for a boolean variable that checks if a user is allowed to do something on your website?
 
 #### --distractors--
 
-To format code automatically.
+`getPermission`
 
 ---
 
-To optimize code for performance.
+`handlePermission`
 
 ---
 
-To create documentation for code.
+`permission`
 
 #### --answer--
 
-To detect potential errors and style issues in code.
+`hasPermission`
 
 ### --question--
 
 #### --text--
 
-Which of the following is NOT a linter for JavaScript?
+Which of the following is best name for a function that runs when a user submits a form?
 
 #### --distractors--
 
-ESLint
+`hasSubmitted()`
```

Comment:
as you are precisely asking for the name, maybe the `()` are out of place as they are not part of the name

```suggestion
`hasSubmitted`
```